oscap_report/cli.py

```diff
@@ -1,8 +1,133 @@
+    "to continue running.\n"
+)
+MASSAGE_FORMAT = '%(levelname)s: %(message)s'
+
+
+class CommandLineAPI():
+    def __init__(self):
+        self.arguments = self._parse_arguments()
+        self.log_file = self.arguments.log_file
+        self.log_level = self.arguments.log_level
+        self._setup_logging()
+        logging.debug("Args: %s", self.arguments)
+        self.report_file = self.arguments.FILE
+        self.output_file_path = self.arguments.output
+        self.output_format = self.arguments.format.upper()
+
+    def _parse_arguments(self):
+        parser = argparse.ArgumentParser(
+            prog="oscap-report",
+            formatter_class=argparse.RawTextHelpFormatter,
+            description=DESCRIPTION,
+        )
+        self._prepare_arguments(parser)
+        return parser.parse_args()
+
+    @staticmethod
+    def _prepare_arguments(parser):
+        parser.add_argument(
+            'FILE',
+            type=argparse.FileType("r"),
+            nargs='?',
+            default=stdin,
+            help="ARF file, stdin if not provided")
+        parser.add_argument(
+            "--version",
+            action="version",
+            version="%(prog)s " + __version__)
+        parser.add_argument(
+            "-o",
+            "--output",
+            action="store",
+            type=argparse.FileType("w+"),
+            default=stdout,
+            help="write the report to this file instead of standard output.")
+        parser.add_argument(
+            "--log-file",
+            action="store",
+            default=None,
+            help="if not provided - stderr")
+        parser.add_argument(
+            "--log-level",
+            action="store",
+            default="WARNING",
+            choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+            help=(
+                "creates LOG_FILE file with log information depending on LOG_LEVEL."
+                "\n{}").format(LOG_LEVES_DESCRIPTION)
+        )
+        parser.add_argument(
+            "-f",
+            "--format",
+            action="store",
+            default="html",
+            help="FORMAT: html, Default: html"
+        )
+
+    def _setup_logging(self):
+        numeric_level = getattr(logging, self.log_level.upper(), None)
+        if not isinstance(numeric_level, int):
+            raise ValueError("Invalid log level: {}".format(self.log_level))
+        logging.basicConfig(
+            format=MASSAGE_FORMAT,
+            filename=self.log_file,
+            filemode='w',
+            level=numeric_level
+        )
+
+    def generate_report(self, report_parser):
+        logging.info("Generate report")
+        report_generator = ReportGenerator(report_parser)
+        if self.output_format == "HTML":
+            return report_generator.generate_html_report()
+        return None
+
+    def load_file(self):
+        logging.info("Loading file")
+        logging.debug(self.report_file)
```

Comment:
It's good to pair run-time values with static text. In that case searching in source code for `Loading file` would bring you right where it was emitted. Also I see no sense in having these two messages on different level.

Like `logging.info("Loading file: %s", self.report_file)`